apps/cowswap-frontend/src/legacy/state/price/updater.ts

```diff
@@ -152,6 +153,9 @@ export default function FeesUpdater(): null {
   const isUnsupportedTokenGp = useIsUnsupportedTokenGp()
 
   const permitHookParams = usePermitHookParams(sellCurrency)
+  const permitHookData = usePermitHookData(permitHookParams)
+  useUpdateAppDataHooks(permitHookData ? { pre: [permitHookData] } : undefined)
+  const appData = useAppData()
```

Comment:
This doesn't look good.

Should probably encapsulate this away somewhere else. Suggestions?

What I want to do this:
1. When the sell token is set and the token is permittable, generate the permit hook
2. Add the generated permit hook to the appData
3. Use the generate appData in the quote request

Reply:
As mentioned by Sasha during the call, here's the auto generated modules diagram for this file 
![price:updater](https://github.com/cowprotocol/cowswap/assets/43217/200753fd-9ee9-41fa-be26-afbd2216ba3b)

(Also in [Mermaid](https://mermaid.live/edit#pako:eNqlVl1v2jAU_Ssoz61EaGkpD5PSEjY0utJB25dIk4lvwMOxI9sZQlX_e00gqW1c2mo84ZPj-3nuTZ6DlGMI-kFKkZQDghYC5QnDRECqCGet2XXCqmctpolh6zlhLf27ExjET8Jwwl5MQqdTM0oJcZZpK83xFnLu0EOT_oQoBTVimcvqGqSoKAZIoeb8UGCkavQH5yvpRnRW3x7Ew-hhPPsziG9Gt9F46hAvDC-PIEhGAN-XXIGMGZpTcFMNe_WFBaiJICncl4gStXED6Fq3Lq1TaJaL8XKxvEYUsRQihiNK-Xr73zF4ZdwZyZijWC2HmurQ2ubhzPJqpvobMlDpssr0RndgjtKVY-mypo9hgdLNEKBiT5DWilvu8NyK7oHJsii4UIBnfAXse-HyzeZOQOREbbto9fgN9rrsmfKgdNczI4ZrkKoCxxxhwhbNoyns8FgILty4mipXlK0qRY62I3E3_1up2m7y-fsq26JDAhS_X8RhSenBYB1z7pXdFiQywliAlPXxSaCi0FnPBMKwA5XYaLcSbkohgKWbKOclU0dq6A7dAOb6RgoGIMg_wNM1KnbzW0v619QK53ApHBH9xw201d-Ad4wSBh-osHn4pIvN149EEj3kzrLaH6qmPOoq6-tfF-vRQfNLsUZ1PadK77ej6-74zvpouXbatXL3GrXHoG0M13tdNsK0LjczcaBCOwJzCdqltnlv5hpR2cHqLZew3XuqdXr6rVWP5g7U76ZPo6EX7X4a1G8dD3rhddXzGvCaDS-9qDfdKx_Y9oHeWENvsH7_5160-3m0939g6M214002_ALqzavjF5fX6kFlg5Mg1wsCEay_uSo5J4FaQg5J0Nd_MWSopCoJtLI1FZWKTzcsDfpKlHASlNX077_SbDDGRHER9DNEJby8AtuRRtc) although doesn't look so nice)

Reply:
There's something wrong with this as, in the first few requests, the quote does not contain the hooks data.
Which in turn makes the order placement request to fail.

Reply:
I don't see the depencency between permit and the price updater. Price updater should be dumb and not care about specific metadata